hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -484,8 +484,20 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		ObservedGeneration: nodePool.Generation,
 	})
 
+	// Initialize NodePool annotations
+	if nodePool.Annotations == nil {
+		nodePool.Annotations = make(map[string]string)
+	}
+
+	// Retrieve pull secret to check for changes when config is checked for updates
+	pullSecretBytes, err := r.getPullSecretBytes(ctx, hcluster)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	pullSecret := string(pullSecretBytes[:])
+
 	// Check if config needs to be updated.
-	targetConfigHash := hashStruct(config)
+	targetConfigHash := hashStruct(config + pullSecret)
```

Comment:
I wonder if as a principle we should consider the content of the secret immutable so for us to trigger and update the referenced name must be different? thoughts? cc @csrwng

Reply:
I understand why we would want to do this. I'm not exactly sure about the implementation. 

If it's based on the Secret's reference name being different, couldn't someone still change the content of the original Secret or delete the original Secret altogether without causing any issues with the guest cluster?

Reply:
@enxebre I agree. We follow this pattern in other parts of the API (secrets/configmaps referenced from the HostedCluster configuration for example). 
@bryan-cox Yes, you can change the content of the secret or even delete the secret, but it's not a change that we're watching for, nor do I think that we should watch for. It's up to the API consumer to supply the proper reference. We only watch for changes in the NP resource.

Reply:
Ok, I've updated the code to look at the pull secret name now instead of the contents of the pull secret.